test/vcf/debugulator_test.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015-2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/filesystem.hpp>
+
+#include "catch/catch.hpp"
+
+#include "parser_test_aux.hpp"
+#include "vcf/debugulator.hpp"
+#include "vcf/report_writer.hpp"
+#include "vcf/sqlite_report.hpp"
+
+namespace ebi
+{
+  bool is_valid_vcf_file(std::istream &input,
+                         ebi::vcf::Parser &validator,
+                         std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
+  {
+      std::vector<char> line;
+      line.reserve(default_line_buffer_size);
+
+      while (ebi::util::readline(input, line)) {
+          validator.parse(line);
+
+          for (auto &error : *validator.errors()) {
+              for (auto &output : outputs) {
+                  output->write_error(*error);
+              }
+          }
+          for (auto &error : *validator.warnings()) {
+              for (auto &output : outputs) {
+                  output->write_warning(*error);
+              }
+          }
+      }
+
+      validator.end();
+
+      return validator.is_valid();
+  }
+
+  TEST_CASE("Fixing a VCF with duplicates", "[debugulator]")
```

Comment:
System tests could be extracted to a different file than unit tests. That would remove redundancy, because there is a whole-process test for the validator side too, and it feels a bit redundant now.